File: L/2264-largest-3-same-digit-number-in-string/main.go
package main

func largestGoodInteger(num string) string {
	n := byte(' ')
	cnt := 0
	b := byte(' ')
	for i := 0; i < len(num); i++ {
		if num[i] == b {
			cnt++
			if cnt >= 3 && n < b {
				n = b
			}
		} else {
			b = num[i]
			cnt = 1
		}
	}
	if n == byte(' ') {
		return ""
	}
	return string([]byte{n, n, n})
}

File: L/2264-largest-3-same-digit-number-in-string/main.rs
struct Solution {}

impl Solution {
  pub fn largest_good_integer(num: String) -> String {
    let bb = num.as_bytes().to_vec();
    let mut idx: usize = 2;

    let mut c: u8 = b' ';
    while idx < bb.len() {
      if bb[idx - 2] == bb[idx - 1] && bb[idx - 1] == bb[idx] {
        if (c == b' ' || c < bb[idx]) {
          c = bb[idx];
        }
        idx += 3;
      } else {
        idx += 1;
      }
    }

    if c == b' ' {
      "".to_string()
    } else {
      String::from_utf8(vec![c, c, c]).unwrap()
    }
  }
}
